security/pkg/pki/ca/ca.go

```diff
@@ -330,7 +333,15 @@ func updateCertInConfigmap(namespace string, client corev1.CoreV1Interface, cert
 // returns the certificate chain and the private key.
 func (ca *IstioCA) GenKeyCert(hostnames []string, certTTL time.Duration) ([]byte, []byte, error) {
 	opts := util.CertOptions{
-		RSAKeySize: 2048,
+		RSAKeySize: rsaKeySize,
+	}
+
+	// use the type of private key the CA uses to generate a CSR of that type (e.g. CA cert using RSA will
+	// cause CSRs using RSA to be generated)
```

Comment:
This is used for creating the Intermediate CA, update the comments and remove TODO

Reply:
fixed